Redesigned routing and anchoring

JanKoehnlein opened the pull request:
Lots of changes:
- introduced manhattan routing, polyline routing and their common part linear routing
- added a registry for anchor computers
- introduced a routing module
- unified and integrated move, move handles and reconnect commands
- added self edge routing
- fixed undo/redo of all commands
- converted Routable interface into SRoutableElement classs
- collected routing related classes in routing/model
- avoid showing hovers when mouse enters while button is pressed

Missing features will be covered in separate issue.

spoenemann commented on src/features/routing/routing.ts:
```diff
@@ -14,17 +14,44 @@
+    /**
+     * Calculates the route of the given edge.
+     *
+     * @param edge
```
Minor: remove undocumented `@param` tag. Same for other methods in this interface.

JanKoehnlein replied:
Done

JanKoehnlein commented on src/features/decoration/decoration-placer.ts:
```diff
@@ -14,18 +14,22 @@
+import { EdgeRouterRegistry } from "../routing/routing";
 
 @injectable()
 export class DecorationPlacer implements IVNodeDecorator {
```
Filed #56

JanKoehnlein commented on src/features/hover/hover.ts:
```diff
@@ -129,11 +129,24 @@ export interface HoverState {
 
 export abstract class AbstractHoverMouseListener extends MouseListener {
+
+    protected mouseIsDown: boolean;
```
Having read the (docs)[https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button]
> This property only guarantees to indicate which buttons are pressed during events caused by pressing or releasing one or multiple buttons. As such, it is not reliable for events such as mouseenter, mouseleave, mouseover, mouseout or mousemove.
I decided to got the other way.

JanKoehnlein commented:
Thanks a lot for the review. Here's to your points:
> Class Diagram Example: Create a polyline route, then drag an anchor => route is deleted.

That's intended. At least for manhattan routing it doesn't make sense to keep the existing route when reconnecting an edge. To be discussed.

> Class Diagram Example: Start dragging the polyline anchor at "Bar", move it below the node, wait a bit until the hover popup appears _below the mouse cursor_, release mouse button => edge is left with a dangling anchor. 

Was a missing guard in `HoverMouseListener#mouseMove()`. Fixed

> Circles example: Performance is very bad when moving nodes manually.
> Circles example: Strange red artifacts appear in the top left corner when moving nodes manually.

Actually both are the same problem: We updated the edge handles when source/target are moved, regardless of whether the edge was editable or not. The red thingies and poor performance were a follow up as we haven't registered views for the routing handles. Fixed
